chore: Remove artifacts of removed commands

Glutexo opened the pull request:
Removed what remained after removing obsolete commands.

* `attach` and partially `--auto_attach` in https://github.com/candlepin/subscription-manager/pull/3409
* rest of `--auto_attach` and `autodial` in https://github.com/candlepin/subscription-manager/pull/3451
* `remove` in https://github.com/candlepin/subscription-manager/pull/3444
* `addons`, `role`, `service-level`, and `usage` in https://github.com/candlepin/subscription-manager/pull/3417
* `redeem` in https://github.com/candlepin/subscription-manager/pull/3446
* `import` in https://github.com/candlepin/subscription-manager/pull/3425

There is more obsolete code to be removed, e.g. in `EntitlementDBusObject`. That’d be over the scope of this pull request, though.

Card IDs:

* [CCT-603](https://issues.redhat.com/browse/CCT-603)

Glutexo commented on src/subscription_manager/entcertlib.py:
```diff
@@ -108,10 +108,6 @@ def perform(self) -> "EntCertUpdateReport":
         self.syslog_results()
 
-        # We call EntCertlibActionInvoker.update() solo from
-        # the 'attach' cli instead of an ActionClient. So
```
I removed the comment, because “'attach' cli” seemed to refer to the CLI command.

ptoscano replied:
except the 'attach' reference, the rest of the comment is actually valid, and dropping it would make the below `refresh()` call without explanation

Glutexo replied:
Ok! I‘ll put the comment back except the `attach` command reference.

Glutexo commented on src/rhsmlib/client_info.py:
```diff
@@ -13,8 +13,8 @@
 
 """
 This module holds information about current state of client application
-like sender of D-bus method, current subscription-manager command (register,
```
Removed `register` too, so it’s not the only example. I was not sure what other command to put there instead of `attach`. There is no example of a DNF command too.

ptoscano replied:
keeping "register", even as the only example should be fine

Glutexo replied:
I put a reference to the `status` command there. It sounds like the best command example after `register`.

Glutexo commented:
I created this pull request as a draft, thinking that I’d put in the command removals as individual commits. If @pinotree approves, however, I think we can merge and do the other commands in (a) new pull request(s).

Glutexo commented:
@ptoscano Thank you for the review! 🙇🏻‍♂️ I addressed your comments.

I un-drafted the PR, so it can be merged if you approve of the changes. There are already 10 changed files. That’s enough for a single PR.

Glutexo commented:
Matyáš suggested we can be more aggressive removing code that seems unused. Under that suggestion, I removed some methods from `EntitlementService` and `EntitlementDBusObject` that doesn’t seem to be called or referenced from anywhere.

Glutexo commented:
Added https://github.com/candlepin/subscription-manager/pull/3476/commits/fda942a5cc4c3fb69a3c2a1196a88067b1fd55a3 removing the artifacts of the `redeem` command.

Glutexo commented:
Added https://github.com/candlepin/subscription-manager/pull/3476/commits/516d70b51b53f2f6224781b55181dcae2e39a459 removing artifacts of the `import` command.

Glutexo commented:
`bindByEntitlementPool` and `bind` weren’t used anywhere after the `attach` command disappeared. Removed.

Glutexo commented:
I merged all the “chore: remove artifacts” pull requests into this one. Removal of every command/option artifacts is in an individual commit. That should make a review easier even though there are many changed files and lines.

I only removed artifacts remaining after what’s been removed in the referenced removal pull requests. I went through their code once more to remove deeper artifacts: methods called by methods (…) called by the removed methods.

* https://github.com/candlepin/subscription-manager/pull/3409
* https://github.com/candlepin/subscription-manager/pull/3451
* https://github.com/candlepin/subscription-manager/pull/3444
* https://github.com/candlepin/subscription-manager/pull/3417
* https://github.com/candlepin/subscription-manager/pull/3446
* https://github.com/candlepin/subscription-manager/pull/3425

There is probably more obsolete code to be removed. An example is the `EntitlementDBusObject`: all its methods except `GetStatus` are not called from anywhere and aren’t referenced in the DBus conf file. I’d keep that for a separate card and pull request though.

I’d like to ask for a real and full review, now. @ptoscano, please. 🙏🏻

ptoscano reviewed (approved):
Thanks!
